src/common/plugin/Artifacts.js

```diff
@@ -0,0 +1,35 @@
+/*globals define*/
+define([
+], function(
+) {
+    const Artifacts = function () {
+        throw new Error('Aritifacts is supposed to be used as an extension object, do not instantiate');
+    };
+
+    Artifacts.prototype.constructor = Artifacts;
+
+    Artifacts.prototype.getBaseNode = function (baseName) {
```

Comment:
I don't think adding a parameter (which is always `"Data"`) is the right solution here. This could be renamed to something like `getArtifactMetaNode` though I think a better solution would be to change the method to perform the entire check that is duplicated in both plugins. Something like `ensureCompatibleMeta` which is an async method that throws the an error with the message returned in the callback. Then each plugin could run that first in their main function.

Reply:
This could mean that we are just adding a new method btw. This method doesn't need to be removed as it could still be used by the other methods in this file. That said, it should still probably be renamed :)